### PR TITLE
Resolves felt/geo#193

### PR DESCRIPTION
**Pull request by carstenpiepel**
Adds a conditions to the Geo.JSON.Decoder module and a corresponding unit test to support Feature objects with a geometry of type GeometryCollection.

### CODE REVIEW

**Comment by devBaqar** [comment]
You don't have a plan to merge this?

File: test/geo/json_test.exs [hunk 1]
```diff
@@ -473,4 +473,57 @@ defmodule Geo.JSON.Test do
...
+          "radarStation": null
+      }
+    }
+    """
+
+    geom = Jason.decode!(json) |> Geo.JSON.decode!()
+
+    assert(Enum.count(geom.geometries) == 2)
+    assert(Enum.each(geom.geometries, fn (%Geo.MultiPolygon{} = g) -> true end))
+    assert geom.properties["id"] == "FLC017"
```
Comment (s3cur3) [inline test/geo/json_test.exs:527]:
```suggestion
    assert %GeometryCollection{} = geom
    assert length(geom.geometries) == 2
    assert Enum.all?(geom.geometries, &match?(%Geo.MultiPolygon{}, &1))
    assert geom.properties["id"] == "FLC017"
```

File: lib/geo/json/decoder.ex [hunk 2]
```diff
@@ -197,7 +197,19 @@ defmodule Geo.JSON.Decoder do
...
+      Map.get(geometry, "type") == "GeometryCollection" ->
+        geometry_collection = decode!(geometry)
+
+        %GeometryCollection{
+          geometries: geometry_collection.geometries,
+          properties: properties
+        }
+
+      true ->
+        do_decode(Map.get(geometry, "type"), Map.get(geometry, "coordinates"), properties, nil)
```
Comment (s3cur3) [inline lib/geo/json/decoder.ex:211]:
```suggestion
    if geometry["type"] == "GeometryCollection" do
      geometry_collection = decode!(geometry)
      
      %GeometryCollection{
        geometries: geometry_collection.geometries,
        properties: properties
      }
    else
      do_decode(Map.get(geometry, "type"), Map.get(geometry, "coordinates"), properties, nil)
```

**Review by s3cur3 — APPROVED** [review]
Thanks so much for your work here, @carstenpiepel! Yep, this makes sense to me, and the test looks good!

File: lib/geo/json/decoder.ex [hunk 3]
```diff
@@ -197,7 +197,16 @@ defmodule Geo.JSON.Decoder do
   defp do_decode("Feature", nil, _properties, _id), do: nil
 
   defp do_decode("Feature", geometry, properties, _id) do
-    do_decode(Map.get(geometry, "type"), Map.get(geometry, "coordinates"), properties, nil)
+    if geometry["type"] == "GeometryCollection" do
+      geometry_collection = decode!(geometry)
+
```
Comment (s3cur3) [inline lib/geo/json/decoder.ex:202]:
```suggestion

```

File: lib/geo/json/decoder.ex [hunk 4]
```diff
@@ -197,7 +197,15 @@ defmodule Geo.JSON.Decoder do
   defp do_decode("Feature", nil, _properties, _id), do: nil
 
   defp do_decode("Feature", geometry, properties, _id) do
-    do_decode(Map.get(geometry, "type"), Map.get(geometry, "coordinates"), properties, nil)
+    if geometry["type"] == "GeometryCollection" do
+      geometry_collection = decode!(geometry)
```
Comment (s3cur3) [inline lib/geo/json/decoder.ex:201]:
```suggestion
      geometry_collection = decode!(geometry)

```

**Pull request merged**